### PR TITLE
fix(fetch-public-farm-data): Increase timeout on farmFetch to 9s

### PR DESCRIPTION
**Pull request by ChefHutch**
- Increase time before dropping farm fetch requests from 5s to 9s
- Add multiplier to return type

### CODE REVIEW

**Review by Chef-Cheems — APPROVED** [review]

**Comment by memoyil** [comment]
Will user have to wait 9 seconds to get the already fetched farm details? If we keep it at 5 (since we are using fast refresh which is 10sec interval) if one farms gets stuck user will only wait 5 seconds and stuck one will be shown in the next cycle which is after 5 seconds.

**Comment by Chef-Cheems** [comment]
@memoyil the farm fetching interval was reduced to slowRefresh (60s) last week (except CAKE - BNB and BUSD - BNB farms). Seems that on weekend there was some issue with farms loading, presumably since most farms took more than 5 sec to load. Hence this PR 🙂

**Comment by memoyil** [comment]
> @memoyil the farm fetching interval was reduced to slowRefresh (60s) last week (except CAKE - BNB and BUSD - BNB farms). Seems that on weekend there was some issue with farms loading, presumably since most farms took more than 5 sec to load. Hence this PR 🙂

What will happen if it takes more than 9 sec? Should we add kind of dynamic variable there? I mean if we remove the timeout we have to wait until the last farm data is fetched which is also not good.

**Comment by ChefHutch** [comment]
So yea Memoyil, just to reiterate what @Chef-Cheems said. I've done some digging and I think our initial diagnosis is right: the timeout of 5s was just too short for many users to complete the request and so the timeout was resolving the promise instead. It would then only attempt the fetch the data again after a slowRefresh.

> What will happen if it takes more than 9 sec? Should we add kind of dynamic variable there? I mean if we remove the timeout we have to wait until the last farm data is fetched which is also not good.

At the moment if 9s passes and the farm hasn't successfully fetched the data - it'll resolve without any of the data being fetched by the smart contract. Let's see what impact this increase has. As you say - the point of the timeout is to prevent the entire farms promise being held up by 1 farm that has failed to fetch. But, we can't have the timeout be so tight that it's _forcing_ farm requests to skip that would otherwise succeed.

If we need to (as is what I think you're suggesting), we could pass a 'timeout' variable into the function - which could be higher for the few farms on a `fastRefresh`, and longer for the majority of farms on a `slowRefresh`.

File: src/state/farms/fetchPublicFarmData.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ type PublicFarmData = {
   lpTotalSupply?: SerializedBigNumber
   tokenPriceVsQuote?: SerializedBigNumber
   poolWeight?: SerializedBigNumber
+  multiplier?: string
```
Comment (memoyil) [inline src/state/farms/fetchPublicFarmData.ts:18]: what is this :D

Comment (Chef-Cheems) [reply src/state/farms/fetchPublicFarmData.ts:18]: it is returned here but was missing in type, I guess @ChefHutch just fixed it while he's changing this file 🙂

**Review by ChefKai — APPROVED** [review]

File: src/state/farms/fetchPublicFarmData.ts (hunk 1 above)
Comment (memoyil) [reply src/state/farms/fetchPublicFarmData.ts:18]: Ah yes true, I thought it is related to timeout thing

**Pull request merged**